### PR TITLE
Add iox_messaging.proto

### PR DESCRIPTION
**Pull request by Norman-To**

### CODE REVIEW

**Comment by jthomas-geotab** [comment]
Missing copyright and revision history.

File: iox_messaging.proto [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+syntax = "proto3";
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/empty.proto";
+
+/**
+ * This Proto file defines message payloads for the Third party Message Type 0x8C (Protobuf Data Packet from IOX to GO) and Message Type 0x26 (Protobuf Data Packet from GO to IOX).
+ * The protobufs defined here follows a simple pub/sub protocol, where a third party IOX device can subscribe to one of the supported TOPICs (enum Topic) 
+ * and receive the desired information from the GO.
+ * The appropriate master switch needs to be set to use the pub/sub functionality correctly.
```
Comment (Stephen-F) [inline iox_messaging.proto]: Consider using another term for 'master switch' as it is likely meaningless for external integrators.

File: iox_messaging.proto [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+ * 4. How the external device gets published information for subscribed topics: 
+ *      When there is an update to a subscribed topic, the GO device sends the update in an IoxFromGo message with the pub_sub.pub field set.
+ *
+ * 5. To remove a topic from the subscription: The external device needs to send a IoxToGo message with the pub_sub.msg.unsub field set. 
+ *      The Go device responds with an IoxFromGo message with the pub_sub.sub_ack.result field containing a 'SUB_ACK_RESULT_SUCCESS'.
+ *
+ * 6. To clear the entire subscription list: The external device needs to send a IoxToGo message with the pub_sub.msg.clear_subs field set.
+ *      The Go device responds with an IoxFromGo message with the pub_sub.clear_subs_ack.result field containing a 'CLEAR_SUBS_ACK_RESULT_SUCCESS'.
+ *
+ * Note: The an IoxFromGo message with the pub_sub.sub_ack.result field or the pub_sub.clear_subs_ack.result field can contain the source
```
Comment (Stephen-F) [inline iox_messaging.proto]: remove an

File: iox_messaging.proto [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+ *
+ * 6. To clear the entire subscription list: The external device needs to send a IoxToGo message with the pub_sub.msg.clear_subs field set.
+ *      The Go device responds with an IoxFromGo message with the pub_sub.clear_subs_ack.result field containing a 'CLEAR_SUBS_ACK_RESULT_SUCCESS'.
+ *
+ * Note: The an IoxFromGo message with the pub_sub.sub_ack.result field or the pub_sub.clear_subs_ack.result field can contain the source 
+ *      of error when the request cannot be performed successfully.
+ *
+ * Note: The subscription is cleared if the GO or the IOX has lost power or if the IOX is disconnected from the GO device.
+ * 
+ * Note, If the master switch is not enabled:
```
Comment (Stephen-F) [inline iox_messaging.proto]: master switch terminology

File: iox_messaging.proto (hunk 1 above)
Comment (Norman-To) [reply iox_messaging.proto]: Will replace all 'master switch' with 'feature flag'.

File: iox_messaging.proto (hunk 2 above)
Comment (Norman-To) [reply iox_messaging.proto]: Removed.

File: iox_messaging.proto (hunk 3 above)
Comment (Norman-To) [reply iox_messaging.proto]: Will replace 'master switch' by 'feature flag'.

**Pull request merged**